lib/hanami/router.rb

```diff
@@ -637,7 +639,7 @@ def not_allowed(env)
     # @since 2.0.0
     # @api private
     def not_found
```

Comment:
@katafrakt I believe it would be useful to pass `env` to the method signature and than eventually to `@not_found`.

```ruby
def not_found(env)
  @not_found.call(env)
end
```